presto-main/src/main/java/com/facebook/presto/sql/planner/SymbolAllocator.java

```diff
@@ -115,6 +118,8 @@ public Symbol newSymbol(String nameHint, Type type, String suffix)
         if (suffix != null) {
             unique = unique + "$" + suffix;
         }
+        // remove special characters for other special serde
+        unique = DISALLOWED_CHAR_PATTERN.matcher(unique).replaceAll("_");
```

Comment:
I am wondering as a follow-up, should we try to replace all name hints that use such special characteristic?